### PR TITLE
Make the baggage API fully functional in the API

### PR DESCRIPTION
**Pull request by jkwatson**
and get rid of the BaggageManager that was needed previously.

This moves the Baggage implementation to the API, per https://github.com/open-telemetry/opentelemetry-specification/pull/1103, which is now merged.

I could split this up, if desired, into 2 parts:
1) Move the implementation to the API, remove it from the SDK, but keep the BaggageManager in place.
2) Remove the BaggageManager entirely.

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/OpenTelemetry.java [hunk 1]
```diff
@@ -126,17 +125,6 @@ public static Meter getMeter(String instrumentationName, String instrumentationV
...
   }
 
-  /**
-   * Returns a singleton {@link BaggageManager}.
-   *
-   * @return registered manager or default via {@link DefaultBaggageManager#getInstance()}.
-   * @throws IllegalStateException if a specified manager (via system properties) could not be
-   *     found.
-   */
-  public static BaggageManager getBaggageManager() {
```
Comment (anuraaga) [inline api/src/main/java/io/opentelemetry/OpenTelemetry.java:136]: 🎉

File: api/src/main/java/io/opentelemetry/baggage/ImmutableBaggage.java [hunk 2]
```diff
@@ -22,24 +17,29 @@
 @Immutable
 // TODO: Migrate to AutoValue
 // @AutoValue
-class BaggageSdk implements Baggage {
+public class ImmutableBaggage implements Baggage {
```
Comment (anuraaga) [inline api/src/main/java/io/opentelemetry/baggage/ImmutableBaggage.java:20]: Does this need to be public?

**Review by anuraaga — APPROVED** [review]
Mostly good, just one point about visibility - presume since we have an interface, we don't need to make the implementation of baggage public

File: api/src/main/java/io/opentelemetry/baggage/ImmutableBaggage.java (hunk 2 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/baggage/ImmutableBaggage.java:20]: hmm. good point. I have to adjust a silly test, but that's easy enough to do.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/baggage/ImmutableBaggage.java]: fixed

**Comment by anuraaga** [comment]
Thanks! Can you rebase to get the build fix?

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Late review, thanks for this!

**Pull request merged**